packages/gatsby/src/bootstrap/requires-writer.ts

```diff
@@ -333,9 +332,8 @@ export const startListener = (): void => {
      * Start listening to CREATE_SERVER_VISITED_PAGE events so we can rewrite
      * files as required
      */
-    emitter.on(`CREATE_SERVER_VISITED_PAGE`, (): void => {
-      reporter.pendingActivity({ id: `requires-writer` })
-      debouncedWriteAll()
+    emitter.on(`CREATE_SERVER_VISITED_PAGE`, async (): void => {
+      await writeAll(store.getState())
     })
```

Comment:
this being debounced here cause whole lot of problems so I got rid of it - it sure is a trade, but timing issues caused by debounce felt worse to deal with than potentially writing out same file a lot in quick succession